src/components/spreadsheet/custom-columns/custom-columns-nodes-config.tsx

```diff
@@ -19,25 +25,130 @@ const styles = {
     },
 };
 
-type CustomColumnsNodesConfigProps = {
+enum NodesOptionId {
+    CONFIG = 'CONFIG',
+    REFRESH = 'REFRESH',
+}
+
+interface NodesOption {
+    id: NodesOptionId;
+    label: string;
+    action: () => void;
     disabled?: boolean;
-};
+    tooltipMsgId?: string;
+    tooltipMsgValues?: Record<string, any>;
+}
+
+interface CustomColumnsNodesConfigProps {
+    disabled?: boolean;
+    tabIndex: number;
+}
 
-export default function CustomColumnsNodesConfig({ disabled }: Readonly<CustomColumnsNodesConfigProps>) {
+export default function CustomColumnsNodesConfig({ disabled, tabIndex }: Readonly<CustomColumnsNodesConfigProps>) {
     const dialogOpen = useStateBoolean(false);
+    const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+
+    const nodesAliases = useSelector((state: AppState) => state.customColumnsNodesAliases);
+    const currentNode = useSelector((state: AppState) => state.currentTreeNode);
+    const tableType = useSelector((state: AppState) => state.tables.definitions[tabIndex].type);
+    const studyUuid = useSelector((state: AppState) => state.studyUuid);
+    const currentRootNetworkUuid = useSelector((state: AppState) => state.currentRootNetworkUuid);
+
+    const { loadEquipmentData } = useLoadEquipment(
+        tableType,
+        studyUuid,
+        currentRootNetworkUuid,
+        currentNode?.id,
+        undefined
+    );
+
+    const nodesToReload = useMemo(() => {
+        // Get all aliased nodes ids, except for Root and current node (both are always up-to-date)
+        return nodesAliases.filter((node) => node.id !== currentNode?.id && node.name !== ROOT_NODE_LABEL);
+    }, [currentNode?.id, nodesAliases]);
```

Comment:
I think this logic can be put in the hook too so that in the end we just call a simple loadEquipmentData() method

Reply:
the nodes list to load/reload are not the same, thats why I pass them to the hook method.

If we want to hide this, I think I need two distinct methods:

- loadEquipmentData()
- reloadEquipmentData()

?

Reply:
I use redux action now